linkerd/app/gateway/src/gateway.rs

```diff
@@ -54,11 +59,30 @@ where
         match self {
             Self::Outbound {
                 ref mut outbound,
+                ref local_identity,
                 ref forwarded_header,
             } => {
+                // Check forwarded headers to see if this request has already
+                // transited through this gateway.
+                for fwd in request
+                    .headers()
+                    .get_all(http::header::FORWARDED)
+                    .into_iter()
+                    .filter_map(|h| h.to_str().ok())
+                {
+                    if let Some(by) = fwd_by(fwd) {
+                        if by == local_identity.as_ref() {
+                            tracing::info!(fwd, "Loop detected");
+                            return rsp(http::StatusCode::LOOP_DETECTED);
```

Comment:
my understanding is that 508 Loop Detected is specific to WebDAV ([per MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/508)). the *name* of the status is definitely correct in this case, but the documented semantics of the code seem to mean something entirely different. is it correct to overload the meaning of that status here, or should we use a different error code?

Reply:
Good point!

I think that leaves us with 502. I'll do that and upgrade the info to a warn.

Reply:
Leaving this as 508 for now, as it's more descriptive to users. I'm less worried about the WebDAV-origin, because it's semantically correct and we're not violating any part of the RFC.